File: challenges/codeeval/medium/071/alealvarezarg.py

import sys
test_cases = open(sys.argv[1], 'r')
for line in test_cases:
    
    line = line.strip()
    
    group = line.split(";")[1]
    numbers = line.split(";")[0].split(",")

    nuevo = []

    count = 0
    tam = len(numbers) // int(group)
    while(count<tam):
        ini = count*int(group)
        fin = (count + 1)*int(group)
        nuevo = nuevo + list(reversed(numbers[ini:fin]))
        count = count + 1

    if(len(nuevo) != len(numbers)):
        pos = count*int(group)
        nuevo = nuevo + numbers[pos::]

    cadena = ",".join(nuevo)

    print(cadena)

test_cases.close()

File: challenges/codeeval/medium/121/alealvarezarg.rb

substitution = {"a" => "y", "b" => "h", "c" => "e", "d" => "s", "e" => "o", "f" => "c", "g" => "v", "h" => "x", "i" => "d", "j" => "u", "k" => "i", "l" => "g", "m" => "l", "n" => "b", "o" => "k", "p" => "r", "q" => "z", "r" => "t", "s" => "n", "t" => "w", "u" => "j", "v" => "p", "w" => "f", "x" => "m", "y" => "a", "z" => "q"}

File.open(ARGV[0]).each_line do |line|
    
    line = line.strip!
    
    count = 0
    cadena = ""
    while(count<line.length)
        if(line[count] != " ")
            cadena = cadena + substitution[line[count]]
        else
            cadena = cadena + " "
        end
        count = count + 1
    end

    puts cadena
    
end
